### ISSUE TITLE
Truvari `refine` doesn't complete

### ISSUE THREAD

**Issue by TimD1** [ACEnglish/truvari#184]
Looks like I have to create a new issue (I don't have permission to re-open).

Reopening this issue, but now with Truvari `refine` (Truvari `bench` now succeeds on this input in ~15 minutes, thanks!). I've filtered out all remotely large variants and all inversions as follows:
```
bcftools view \
    -i 'TYPE=="SNP" || (ILEN < 1000 && ILEN > -1000)'\
    pav.all.vcf.gz |
    grep -v "INV" > pav.most.vcf
bgzip -f pav.most.vcf
tabix -p vcf pav.most.vcf.gz
```

I ran Truvari WFA `refine` on the `bench` results, limiting the GIAB-TR regions to `candidate.refine.bed`. I've included my log file here: [pav.log](https://github.com/ACEnglish/truvari/files/13834058/pav.log).

On this attempt, I defined the default `/tmp` directory to be located on an external hard drive. It crashed after 2.5 hours with a `MemoryError`, and used 475GB of `/x/tmp` memory.

On a previous attempt, it hung for a few days after filling the `/tmp` directory with 185GB of data and then the current directory with another 600GB of data. 

All of this data is located in files named `tmp********`, and they appear to be FASTA files auto-generated by samtools or something. How much space is this expected to take, and how long should the analysis run for?

Thanks in advance, sorry for opening so many issues. I'd be happy to help provide any more info to get this working.

_Originally posted by @TimD1 in https://github.com/ACEnglish/truvari/issues/182#issuecomment-1877555333_

**Comment by ACEnglish** [comment]
Those temporary files are created by `pysam.bcftools.consensus`. A recent ticket about that same submodule forced me to reimplement the haplotype consensus creation step. The new procedure doesn't use temporary files (and is faster, which I wasn't expecting). Please install directly from develop and re-try.

**Comment by TimD1** [comment]
Thanks, I'll try this now and let you know how it goes.

**Comment by TimD1** [comment]
Looks like it's been fixed in v4.2.0. Thanks!
